syntax = "proto3";
package bentoml;

import "common.proto";
import "status.proto";
import "deployment.proto";
import "repository.proto";

// BentoML deployment and management service
// Config DeployTarget
// yatai start --deploy-taget-config
// [aws][sagemaker]
// [google]
// [kubecfg]
service Yatai {

  // Common RPC
  rpc HealthCheck (Empty)
    returns (HealthCheckResponse);


  // Deployment Management RPC
  rpc ApplyDeployment(ApplyDeploymentRequest)
    returns (ApplyDeploymentResponse);

  rpc DeleteDeployment(DeleteDeploymentRequest)
    returns (DeleteDeploymentResponse);

  rpc GetDeployment(GetDeploymentRequest)
    returns (GetDeploymentResponse);

  rpc DescribeDeployment(DescribeDeploymentRequest)
    returns (DescribeDeploymentResponse);


  // BentoService Repository RPC
  rpc AddBentoService(AddBentoServiceRequest)
    returns (AddBentoServiceResponse);

  rpc RemoveBentoService(RemoveBentoServiceRequest)
    returns (RemoveBentoServiceResponse);

  rpc GetBentoService(GetBentoServiceRequest)
    returns (GetBentoServiceResponse);

}

message HealthCheckResponse {
  Status status = 1;
}
